Index: tests/lib/CMakeLists.txt
===================================================================
--- a/tests/lib/CMakeLists.txt
+++ b/tests/lib/CMakeLists.txt	2012-07-14 22:10:51.631357086 +0200
@@ -22,7 +22,7 @@
 endforeach(moc_src ${tp_qt_tests_MOC_SRCS})
 
 add_library(tp-qt-tests ${tp_qt_tests_SRCS})
-target_link_libraries(tp-qt-tests ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} telepathy-qt${QT_VERSION_MAJOR})
+target_link_libraries(tp-qt-tests ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTTEST_LIBRARY} telepathy-qt${QT_VERSION_MAJOR})
 
 if(ENABLE_TP_GLIB_TESTS)
     add_subdirectory(glib)
Index: tests/lib/glib-helpers/CMakeLists.txt
===================================================================
--- a/tests/lib/glib-helpers/CMakeLists.txt
+++ b/tests/lib/glib-helpers/CMakeLists.txt	2012-07-14 22:11:25.981686761 +0200
@@ -18,5 +18,6 @@
         ${DBUS_GLIB_LIBRARIES}
         ${QT_QTCORE_LIBRARY}
         ${QT_QTDBUS_LIBRARY}
+        ${QT_QTTEST_LIBRARY}
         telepathy-qt${QT_VERSION_MAJOR})
 endif(ENABLE_TP_GLIB_TESTS)
Index: TelepathyQt/Farstream/CMakeLists.txt
===================================================================
--- a/TelepathyQt/Farstream/CMakeLists.txt
+++ b/TelepathyQt/Farstream/CMakeLists.txt	2012-07-14 22:12:15.202157681 +0200
@@ -43,6 +43,8 @@
         ${QT_QTCORE_LIBRARY}
         ${TELEPATHY_FARSTREAM_LIBRARIES}
         ${TELEPATHY_GLIB_LIBRARIES}
+        ${GOBJECT_LIBRARIES}
+        ${GLIB2_LIBRARIES}
         telepathy-qt${QT_VERSION_MAJOR}
         ${TP_QT_LIBRARY_LINKER_FLAGS})
 
Index: TelepathyQt/Farsight/CMakeLists.txt
===================================================================
--- a/TelepathyQt/Farsight/CMakeLists.txt
+++ b/TelepathyQt/Farsight/CMakeLists.txt	2012-07-14 22:12:54.132528991 +0200
@@ -30,6 +30,8 @@
         ${QT_QTCORE_LIBRARY}
         ${TELEPATHY_FARSIGHT_LIBRARIES}
         ${GSTREAMER_INTERFACE_LIBRARY}
+        ${TELEPATHY_GLIB_LIBRARIES}
+        ${GOBJECT_LIBRARIES}
         telepathy-qt${QT_VERSION_MAJOR}
         ${TP_QT_LIBRARY_LINKER_FLAGS})
 
